File: src/LineEndingNormalizer.cs
namespace Clipman
{
    internal static class LineEndingNormalizer
    {
        public const string Windows = "\r\n";
        public const string Unix = "\n";
        public const string OldMac = "\r";

        public static string ToWindows(string text)
        {
            return Normalize(text, Windows);
        }

        public static string ToUnix(string text)
        {
            return Normalize(text, Unix);
        }

        public static string ToOldMac(string text)
        {
            return Normalize(text, OldMac);
        }

        public static string Normalize(string text, string targetLineEnding)
        {
            if (text == null) return string.Empty;
            if (string.IsNullOrEmpty(targetLineEnding)) targetLineEnding = Windows;

            return text
                .Replace("\r\n", "\n")
                .Replace("\r", "\n")
                .Replace("\u0085", "\n")
                .Replace("\u2028", "\n")
                .Replace("\u2029", "\n")
                .Replace("\n", targetLineEnding);
        }
    }
}

File: ClipmanMac/Sources/Clipman/LineEndingNormalizer.swift
import Foundation

enum LineEndingStyle {
    case windows
    case unix
    case oldMac

    var separator: String {
        switch self {
        case .windows:
            return "\r\n"
        case .unix:
            return "\n"
        case .oldMac:
            return "\r"
        }
    }
}

enum LineEndingNormalizer {
    static func normalize(_ text: String, to style: LineEndingStyle) -> String {
        text
            .replacingOccurrences(of: "\r\n", with: "\n")
            .replacingOccurrences(of: "\r", with: "\n")
            .replacingOccurrences(of: "\u{0085}", with: "\n")
            .replacingOccurrences(of: "\u{2028}", with: "\n")
            .replacingOccurrences(of: "\u{2029}", with: "\n")
            .replacingOccurrences(of: "\n", with: style.separator)
    }
}
